src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java

```diff
@@ -150,6 +154,9 @@
 
     private static final String API_COMMIT_STATUS_PATH = "/rest/build-status/1.0/commits{/hash}";
     private static final Integer DEFAULT_PAGE_LIMIT = 200;
+    private static final int API_RATE_LIMIT_STATUS_CODE = 429;
+    private static final Duration API_RATE_LIMIT_INITIAL_SLEEP = Duration.ofSeconds(5);
+    private static final Duration API_RATE_LIMIT_MAX_SLEEP = Duration.ofMinutes(30);
```

Comment:
#483 retried every 5 seconds, up to 300 times, for a total of 25 minutes of retries. This PR uses exponential backoff, and in the worst case would wait around 27 minutes, but in those 27 minutes it would only retry around 13 times.

I am not sure what settings would make the most sense here, so I just kept roughly the same timeframe from the earlier PR. Perhaps this should be lowered to only a few minutes with non-exponential backoff. Based on my testing, the in-product documentation, and https://confluence.atlassian.com/bitbucketserver0721/improving-instance-stability-with-rate-limiting-1115666697.html, it seems like anything more than a few minutes is probably unlikely to help in practice, unless you have extremely bursty usage since the rate limit quota is refreshed on a per-second basis.

Reply:
LGTM. Given that each server can have its own settings (https://confluence.atlassian.com/bitbucketserver/improving-instance-stability-with-rate-limiting-976171954.html) I think this is a good trade-off.